### PR TITLE
feat: user should be able to select correct network

### PR DESCRIPTION
**Pull request by imanshu03**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

About # (link your issue here)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by adaptivecodeio** [comment]
@imanshu03 the button on the modal is too round can you develop it like the ones in the designs please thanks

File: src/components/WalletModal/index.tsx [hunk 1]
```diff
@@ -274,21 +280,75 @@ export default function WalletModal({
         })
     }
 
+    const handleEthereumNetworkSwitch = () => {
```
Comment (sirpy) [inline src/components/WalletModal/index.tsx:283]: use useCallback correctly in ALL callback functions
@adaptivecodeio this is a basic thing that should come up in your review

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/components/WalletModal/index.tsx (hunk 1 above)
Comment (adaptivecodeio) [reply src/components/WalletModal/index.tsx:283]: noted

**Comment by adaptivecodeio** [comment]
@sirpy changed pushed

**Review by sirpy — APPROVED** [review]

File: src/components/WalletModal/index.tsx [hunk 2]
```diff
@@ -336,8 +396,11 @@ export default function WalletModal({
     }
 
     return (
-        <Modal isOpen={walletModalOpen} onDismiss={toggleWalletModal} minHeight={false} maxHeight={90}>
-            <Wrapper>{getModalContent()}</Wrapper>
-        </Modal>
+        <>
+            <Modal isOpen={walletModalOpen} onDismiss={toggleWalletModal} minHeight={false} maxHeight={90}>
+                <Wrapper>{getModalContent()}</Wrapper>
```
Comment (sirpy) [inline src/components/WalletModal/index.tsx:401]: why is this a function and not a component?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/components/WalletModal/index.tsx (hunk 2 above)
Comment (imanshu03) [reply src/components/WalletModal/index.tsx]: It was previously implemented by some other developers as a function, I have converted it to a functional component. Please review and approve

**Comment by sirpy** [comment]
@adaptivecodeio @imanshu03 
please make sure developer adds screen shots of UI related PRs

**Review by sirpy — APPROVED** [review]
i approve the functional component fix.
i've not tested functionality

**Pull request merged**